internal/pkg/addon/storage.go

```diff
@@ -45,43 +55,62 @@ var storageTemplateFunctions = map[string]interface{}{
 	"toSnakeCase":   template.ToSnakeCaseFunc,
 }
 
-// StorageProps holds basic input properties for addon.NewDDBTemplate() or addon.NewS3Template().
+// StorageProps holds basic input properties for S3Props and DynamoDBProps.
 type StorageProps struct {
 	Name string
 }
 
-// S3Props contains S3-specific properties for addon.NewS3Template().
+// S3Props contains S3-specific properties.
 type S3Props struct {
 	*StorageProps
 }
 
-// NewS3Template creates a new S3 marshaler which can be used to write CF via addonWriter.
-func NewS3Template(input *S3Props) *S3Template {
+// NewWorkloadS3Template creates a marshaler for a workload-level S3 addon.
+func NewWorkloadS3Template(input *S3Props) *S3Template {
```

Comment:
nit: what do you think of dropping the `New` prefix from all these functions so that it's just `addon.WorkloadS3Template`, `addon.EnvS3Template`